agent/discovery/query_fetcher_v2.go

```diff
@@ -283,6 +287,50 @@ func getEndpointWeight(endpoint *pbcatalog.Endpoint, configCtx *v2DataFetcherDyn
 	return weight, true
 }
 
+// getResultPorts conditionally returns ports from a map based on a query. The results are sorted by name.
+func getResultPorts(req *QueryPayload, workloadPorts map[string]*pbcatalog.WorkloadPort) []Port {
+	if workloadPorts == nil {
+		return nil
+	}
+
+	// If a port is specified, make sure the workload implements that port name.
+	if _, ok := workloadPorts[req.PortName]; req.PortName != "" && !ok {
+		return nil
+	}
+
+	var ports []Port
+	if req.PortName != "" {
+		// In the case that the query asked for a specific port, we only return that port.
+		ports = []Port{
+			{
+				Name:   req.PortName,
+				Number: workloadPorts[req.PortName].Port,
+			},
+		}
+	} else {
+		// If the client didn't specify a particular port, return all the workload ports.
+		for name, port := range workloadPorts {
+			ports = append(ports, Port{
+				Name:   name,
+				Number: port.Port,
+			})
+		}
+		if ports == nil {
```

Comment:
i think the only way this would trigger is if `workloadPorts` is nil or empty, which should get handled at the top of the function, so this might be unreachable code.

Reply:
Yeah, with the previous suggestion to use `len` I can remove this now.